src/clients/Deluge.ts

```diff
@@ -25,9 +25,26 @@ enum DelugeErrorCode {
 interface TorrentInfo {
 	complete: boolean;
 	save_path: string;
+	label?: string | undefined;
+}
+interface InjectResponse {
+	result?: boolean;
+	error?: {
+		message?: string;
+	};
+}
+interface TorrentObject {
+	progress: number;
+	save_path: string;
+	state: string;
 	label?: string;
 }
-
+interface InfoResponse {
+	result?: { torrents?: Record<string, TorrentObject> };
+	error?: {
+		message?: string;
+	};
+}
```

Comment:
you can combine these with DelugeResponse if you make DelugeResponse generic
```ts
type InfoResponse: DelugeResponse<{ torrents : Record<string, TorrentObject }>, ErrorType>
```

`ErrorType` might always be the same - an object with `message` and `code`.

if you want to get really fancy, you can make a union:
```ts
type ResultType = { result: string, foo: string }
type ErrorType = { error: string, bar: string}
type DelugeResponse = ResultType | ErrorType;

const x: DelugeResponse = undefined as unknown as DelugeResponse;

if ("error" in x) {
    x.bar
} 
if ("result" in x) {
    x.foo
}

```

Reply:
Something like this should be sufficient I think?